lib/fabrik/DirectorInstance.js

```diff
@@ -53,34 +57,52 @@ class DirectorInstance extends BaseInstance {
         if (operation.type === 'delete') {
           return Promise
             .all([
-              this.deleteSecurityGroup(),
-              this.ensureSpaceGuid()
-              .then(space_guid => this.manager.deleteRestoreFile(space_guid, this.guid))
-              .catch(err => {
-                logger.error(`Failed to delete restore file of instance '${this.guid}'`);
-                logger.error(err);
-                throw err;
-              })
+              this.platformManager.preInstanceDeleteOperations({
+                ruleId: this.securityGroupName
+              }),
+              this.deleteRestoreFile()
               //ScheduledBackupJob is not being cancelled on delete, rather BackupJob itself takes care of it
             ]);
         }
       });
   }
 
+  deleteRestoreFile() {
+    Promise.try(() => {
+      if (_.includes(this.manager.agent.features, 'backup')) {
+        return this.platformManager.ensureTenantId()
+          .then(space_guid => this.manager.deleteRestoreFile(space_guid, this.guid))
+          .catch(err => {
+            logger.error(`Failed to delete restore file of instance '${this.guid}'`);
+            logger.error(err);
+            throw err;
+          });
+      }
+    });
+  }
+
   finalize(operation) {
     const action = _.capitalize(operation.type);
     return Promise
       .try(() => {
         switch (operation.type) {
         case 'create':
-          return this
-            .createSecurityGroup(operation.space_guid)
+          return this.platformManager.postInstanceProvisionOperations({
+              ruleId: this.securityGroupName,
```

Comment:
rule id - security group name , can be within platform manager.

Reply:
removed